.github/workflows/ngen_integration.yaml

```diff
@@ -0,0 +1,119 @@
+  MACOS_NUM_PROC_CORES: 3
+  ASAN_OPTIONS: detect_leaks=false
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # Run general unit tests in linux environment
+  test_surfacebmi_plus_cfe:
+    # The type of runner that the job will run on
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checkout and save the topmodel to a temporary directory
+      - name: Checkout the commit
+        uses: actions/checkout@v4
+
+      - name: Save Topmodel to a Temp Directory
+        run: |
+          # Move files to a temporary directory
+          mkdir ${{runner.temp}}/topmodel
+          mv ./* ${{runner.temp}}/topmodel
+
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: checkout actions in another repo
+        uses: actions/checkout@v4
+        with:
+          repository: noaa-owp/ngen
+
+      - name: Build Surfacebmi
+        id: submod_build_1
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: " extern/noah-owp-modular/"
+          targets: "surfacebmi"
+
+      - name: Build ISO C Fortran BMI
+        id: submod_build_2
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/iso_c_fortran_bmi/"
+
+      - name: Build CFE
+        id: submod_build_3
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/cfe/"
+          targets: "cfebmi"
+
+      - name: Build PET
+        id: submod_build_5
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/evapotranspiration/evapotranspiration"
+          targets: "petbmi"
+
+      - name: Build SLoTH
+        id: submod_build_6
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/sloth/"
+          targets: "slothmodel"
+
+      # Build Ngen and save some files
+      - name: Build Ngen
+        id: ngen_id1
+        uses: ./.github/actions/ngen-build
+        with:
+          targets: "ngen"
+          build-cores: ${{ env.LINUX_NUM_PROC_CORES }}
+          #is this required for this test?
+          bmi_c: 'ON'
+          bmi_fortran: 'ON'
+        timeout-minutes: 15
+
+      - name: Prepare to Build Topmodel Library for Ngen
+        run: |
+          # Move files to appropriate directory to build topmodel library
+          rm -rf extern/topmodel/topmodel/*
+          mv ${{runner.temp}}/topmodel/* extern/topmodel/topmodel
+
+      - name: Build Topmodel
+        id: submod_build_4
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: " extern/topmodel/"
+          targets: "topmodelbmi"
+
+      - name: Run Ngen Test
+        run: |
+          # Run ngen with topmodel with CFE
```

Comment:
Again, this [realization](https://github.com/NOAA-OWP/ngen/blob/master/data/example_bmi_multi_realization_config.json) couples `noahowp` & `cfe`. 

Is a `topmodel` & `CFE` coupling even sensible/common?  @SnowHydrology

It's probably best to throw `PET` somewhere in the mix.  The lib is being built, but not currently used. @stcui007 - I believe I have a `bmi_multi_pet_topmodel` realization around somewhere if that helps. -thanks again.

Reply:
Tha comment is incorrect. I think it is either topmodel or CFE.

Reply:
Could try to see if you can find your realization file coupling PET with Topmodel? That would save me time if there are intricacies involved. Thanks.

Reply:
@stcui007 - See pr I submitted to [your fork](https://github.com/stcui007/topmodel/pull/1).  Once that is (squashed &) merged, this pr here should be good to go. -thanks!